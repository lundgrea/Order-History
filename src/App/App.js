import React, { Component } from 'react';
import OrderContainer from '../OrderContainer/OrderContainer'
import Form from '../Form/Form'
import { postNewOrder } from '../apiCalls'
import './App.css';

class App extends Component {
  constructor() {
    super();
    this.state = {
      orders: [],
      isLoading: true,
      error: ''
    }
  }

  removeOrder = (id) => {
    const filteredOrders = this.state.orders.filter(order => {
      return order.id != id
    })
    this.setState({orders: filteredOrders})
  }

  addOrder = (newOrder)=> {
    postNewOrder(newOrder)
    .then(order => this.setState({orders: [...this.state.orders, newOrder]}))
    .catch(error => this.setState({error : error.message}))
  }

  componentDidMount() {
    fetch('http://localhost:3001/api/v1/purchases')
      .then(response => response.json())
      .then(orders => this.setState({orders: orders}))
      .catch(error => this.setState({error: error.message}))
  }

  render() {
    return (
      <div className="App">
        <header>
          <h1 className='app-title'>My Order History</h1>
          <div className='purchase-form'>
            <Form addOrder={this.addOrder}/>
          </div>
        </header>
        <div className='purchase-container'>
          <OrderContainer orders={this.state.orders} removeOrder={this.removeOrder}/>
        </div>
      </div>
    );
  }
}

export default App;
