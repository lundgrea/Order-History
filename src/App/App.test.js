import React from 'react';
import { shallow } from 'enzyme'
import App from './App';

describe('App', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<App/>)
  })

it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot()
})

it('should update state with an order when addOrder is called', () => {
  const mockOrder = {
    img: 'URL',
    name: 'iphone',
    description: 'one of those phones',
    price: "500",
    id: Date.now()
  }
  const expected = [mockOrder]

  expect(wrapper.state('orders')).toEqual([])
  wrapper.instance().addOrder(mockOrder)
  expect(wrapper.state('orders')).toEqual(expected)
});

})
