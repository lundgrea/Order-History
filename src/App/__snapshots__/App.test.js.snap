// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match snapshot 1`] = `
<div
  className="App"
>
  <header>
    <h1
      className="app-title"
    >
      My Order History
    </h1>
    <div
      className="purchase-form"
    >
      <Form
        addOrder={[Function]}
      />
    </div>
  </header>
  <div
    className="purchase-container"
  >
    <OrderContainer
      orders={Array []}
      removeOrder={[Function]}
    />
  </div>
</div>
`;
